third-culture-software/bhima
# Reimplemented Current Assignments report

Reconnected old Assignments registry report.
- Does not implement a CSV version.  If that is desired, we can add it later.

Although "Assignments" can be applied to all lots, assigning/etc, is only supported via the Assets Management pages.  Hence this report is under that menu.   This is due to earlier decisions regarding handling assets differently than non-asset lots (although it could be changed).  We currently have no hints of assignment-related functionality in the Stock Lots registry.

Although this report can currently be accessed under the Assets Registry Menu, it could be moved under Assets Management > Reports.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6621

**TESTING**
- Use bhima_test
- Go to:  Asset Management > Assets Registry
   - This should show two assets MOT1, and MOT2.  Only MOT1 is assigned
   - Use the Menu > Current Assignments and examine the resulting report
      - It should show only the the assignment for MOT1
   - Using the action menu for MOT2, assign it to an entity
   - Repeat generating the report and verify that both assets are shown
   - Test adding an assignment to a non-asset (eg, drug) using the "+ New Assignment" above the Assets Registry.  
     Even though it will not be shown in the asset registry, it will be shown in the Current Assignments report.
